### ISSUE TITLE
[Bug]: Multiple windows slow down or stop videos

### ISSUE THREAD

**Issue by dvoraking** [FreeTubeApp/FreeTube#2507]
### Guidelines

- [X] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [X] I have searched the [issue tracker](https://github.com/FreeTubeApp/FreeTube/issues) for a bug report that matches the one I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.

### Describe the bug

Opening multiple windows can slow down videos. They either stop, requiring one video to be closed for the other to play, or they crash both, requiring them to both be closed. When this happens, they usually forget the position on the timeline they were saved at. Used to be around 5 could be opened, but sometimes as little as 2 are enough.

### Expected Behavior

I expect to open more than 5 videos and have them run independently, like I would in a browser running Invidious or Newpipe. Now I know there's a good chance they'll get messed up.

### Issue Labels

content not loading

### FreeTube Version

v0.17.1

### Operating System Version

Windows 10 Pro 21H2

### Installation Method

.exe

### Primary API used

Local API

### Nightly Build

- [x] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Is this still an issue on [the latest nightly build](https://docs.freetubeapp.io/development/nightly-builds)?

**Comment by dvoraking** [comment]
Yeah. freetube Setup 0.17.1-nightly-2017.exe. Although now it seems it can run 2 videos fairly comfortably, a 3rd one freezes everything. Then I have to close 2 to get 1 working.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Hmm that is weird. What are the specs of your system?

**Comment by dvoraking** [comment]
R5 3600, 16GB, RTX 3080, Win 10 Pro, PCIE3.0 SSD. Tested it again. 2 videos can be played, which is better than the stable release, but a 3rd freezes them. How many can you get playing in one go?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Hmm it cant be your system. Im playing 10+ and thats on a low spec laptop

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Hi there, we just release v0.18.0. Is this still an issue for u on v0.18.0?

**Comment by dvoraking** [comment]
For a brief moment, it worked. After updating, I got 8 or 9 videos working. Then, the next day, it suddenly went back to 3.

**Comment by dvoraking** [comment]
I'm watching the latest video from a 1m subscriber channel within the first hour, and it tanked the whole thing. I can't even play 2 videos at once. I'm better off using my browser. A similar thing happens with music videos on Invidious, except they just won't play most of the time. I heard it's the popularity of music videos, and the number of people trying to watch it at the same time. Maybe that affects Freetube as it uses Invidious' instance.

**Comment by absidue** [comment]
Do you have the proxy videos through Invidious option enabled or disabled?

**Comment by dvoraking** [comment]
Disabled. I'll enable it and see what difference it makes. Although I've noticed it usually falls back on Invidious anyway.

**Comment by absidue** [comment]
@dvoraking which version of FreeTube are you using?

**Comment by dvoraking** [comment]
freetube-0.18.0-nightly-2664-setup-x64.exe

**Comment by dvoraking** [comment]
My internet speed more than doubled, nearly tripled thanks to a new contract. Still the same issues. Usually 3, sometimes just 1, sometimes 5-6 open. Latest nightly build (Translated using Weblate (German) #2758). Don't think it's an internet issue, and we've established the hardware is fast enough.

**Comment by aleksejrs** [comment]
With a recent nightly, if I open many FreeTube windows (37 now), FreeTube stops working.  Xfce panel shows the window buttons only for a moment after leaving the workspace.

**Comment by absidue** [comment]
@aleksejrs sounds like you are overwhelming your computer with that many instances open. Not sure it's really a FreeTube issue at that point anymore, if are doing more than your computer can handle it. Is there actually a legitimate use case for using that many windows, or where you trying to break it on purpose?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
The Author has this issue with a minimum of 3 windows open so commenting with something like 40 on this issue doenst make sense.

**Comment by aleksejrs** [comment]
@absidue, if it's "overwhelming my computer", it could be a hardware issue, because freeing CPU and RAM didn't make FreeTube come back. But hardware issues haven't happened for months with the current setup (CPU is throttled based on temperature and not allowed to run at the max frequency, because something had been overheating and causing various glitches, so the frequent 100% load does not lead to issues anymore).

It was the second time I opened many FreeTube windows to collect links for downloading, and it hung unrecoverably.

Maybe it's overwhelming Xfce? But then why do FreeTube's windows become unavailable and Xfce suggests killing it?

**Comment by aleksejrs** [comment]
I filed #3671 for my issue of FreeTube hanging with 35-50 videos open.

**Comment by Xelbayria** [comment]
```
Local API Error (click to copy): Error: Status code: 410
Falling back to Invidious API
```
I am seeing the same error (above) when i am playing a video via Freetube.  The problem is the way video play... Here's the picture to point out the problem: 
![image](https://github.com/FreeTubeApp/FreeTube/assets/12866666/d804aa51-79b3-41a6-b951-dfc64a50039f)

U should be able to see "green" which i am currently at, If i keep playing the video, then I'll catch up to the red arrow (idk what do you call that part). it's the grey part that the video is being load ahead of time. 

it's very extremely slow and sometimes I caught up and it take forever to load. 

I am also using v0.18.0-Beta and it doesn't matter which video. I also tested the browser, i had no problem. Is there something wrong with Invidious?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Hi @dvoraking do you still see this issue in v0.24.0?
